packages/vaadin-virtual-list/src/virtualizer-iron-list-adapter.js

```diff
@@ -0,0 +1,430 @@
+      this.scrollTarget.addEventListener('mousedown', () => (this.__mouseDown = true));
+      this.scrollTarget.addEventListener('mouseup', () => {
+        this.__mouseDown = false;
+        if (this.__pendingReorder) {
+          this.__reorderElements();
+        }
+      });
+    }
+  }
+
+  _manageFocus() {}
+
+  _removeFocusedItem() {}
+
+  get scrollOffset() {
+    return 0;
+  }
+
+  scrollToIndex(index) {
+    if (typeof index !== 'number' || isNaN(index) || !this.scrollTarget.offsetHeight) {
+      return;
+    }
+    index = this._clamp(index, 0, this.size - 1);
+
+    const visibleElementCount = this.__getVisibleElements().length;
+    let targetVirtualIndex = Math.floor((index / this.size) * this._virtualCount);
+    if (this._virtualCount - targetVirtualIndex < visibleElementCount) {
+      targetVirtualIndex = this._virtualCount - (this.size - index);
+      this._vidxOffset = this.size - this._virtualCount;
+    } else if (targetVirtualIndex < visibleElementCount) {
+      if (index < OFFSET_ADJUST_MIN_THRESHOLD) {
+        targetVirtualIndex = index;
+        this._vidxOffset = 0;
+      } else {
+        targetVirtualIndex = OFFSET_ADJUST_MIN_THRESHOLD;
+        this._vidxOffset = index - targetVirtualIndex;
+      }
+    } else {
+      this._vidxOffset = index - targetVirtualIndex;
+    }
+
+    do {
+      this.__skipNextVirtualIndexAdjust = true;
+      super.scrollToIndex(targetVirtualIndex);
+      this._scrollHandler();
+    } while (this.firstVisibleIndex !== index - this._vidxOffset && this._scrollTop < this._maxScrollTop && !this.grid);
+  }
+
+  flush() {
+    this._resizeHandler();
+    flush();
+    this._scrollHandler();
+    this.__scrollReorderDebouncer && this.__scrollReorderDebouncer.flush();
+    this.__debouncerWheelAnimationFrame && this.__debouncerWheelAnimationFrame.flush();
+  }
+
+  update(startIndex = 0, endIndex = this.size - 1) {
+    this.__getVisibleElements().forEach((el) => {
+      if (el.__virtualIndex >= startIndex && el.__virtualIndex <= endIndex) {
+        this.updateElement(el, el.__virtualIndex);
+      }
+    });
+  }
+
+  __getIndexScrollOffset(index) {
+    const element = this.__getVisibleElements().find((el) => el.__virtualIndex === index);
+    return element ? this.scrollTarget.getBoundingClientRect().top - element.getBoundingClientRect().top : undefined;
+  }
+
+  set size(size) {
+    if (size === this.size) {
+      return;
+    }
+
+    let fvi = this.firstVisibleIndex + this._vidxOffset;
+
+    const fviOffsetBefore = this.__getIndexScrollOffset(fvi);
+
+    this.__size = size;
+    this._itemsChanged({
+      path: 'items'
+    });
+
+    fvi = Math.min(fvi, size - 1);
+    this.scrollToIndex(fvi);
+
+    const fviOffsetAfter = this.__getIndexScrollOffset(fvi);
+    if (fviOffsetBefore !== undefined && fviOffsetAfter !== undefined) {
+      this._scrollTop += fviOffsetBefore - fviOffsetAfter;
+    }
```

Comment:
By default, the _itemsChanged invocation above would always reset the scrolltop back to 0. This logic is there to restore the scroll position on size change. Logic moved from [here](https://github.com/vaadin/web-components/blob/b163d3a8c200e6a9064e03b7ca1c4a34ebb4f3d9/packages/vaadin-grid/src/vaadin-grid-scroller.js#L124-L128)